File: src/main/java/com/staymate/service/StudentService.java
package com.staymate.service;

import java.util.List;

import com.staymate.entity.Student;

public interface StudentService {

    Student saveStudent(Student student);

    List<Student> getAllStudents();

    Student getStudentById(Integer studentId);

    Student updateStudent(Integer studentId, Student student);

    void deleteStudent(Integer studentId);
    
//    StudentResponseDTO createProfile(Long userId, StudentRequestDTO dto);
//
//    StudentResponseDTO getProfile(Long userId);
//
//    StudentResponseDTO updateProfile(Long userId, StudentRequestDTO dto);
}

File: frontend/src/services/studentService.js
import api from "./api";

export const getStudents = () => api.get("/students");

export const getStudentById = (id) =>
    api.get(`/students/${id}`);

export const addStudent = (student) =>
    api.post("/students", student);

export const updateStudent = (id, student) =>
    api.put(`/students/${id}`, student);

export const deleteStudent = (id) =>
    api.delete(`/students/${id}`);